yjit/src/backend/x86_64/mod.rs

```diff
@@ -243,6 +243,16 @@ impl Assembler
                 // Load effective address
                 Op::Lea => lea(cb, insn.out.into(), insn.opnds[0].into()),
 
+                // Load relative address
+                Op::Lera => {
+                    let disp = match insn.opnds[0] {
```

Comment:
It feels more natural to me to integrate the IR instruction with labels rather than asking users to pass in a raw offset. If you're writing IR you don't really have control over lowering and the exact number bytes after the fact. On x64 RIP rel is relative to the end of the current instruction, so if we ask people to put an offset they at least have to know how long the LEA is, which is variable.

Reply:
Alan is right. It's also relative to the end of the current instruction on Arm, but this would definitely be more robust using labels.